src/chunk.c

```diff
@@ -18,71 +17,77 @@ Chunk * NewChunk(size_t sampleCount)
     return newChunk;
 }
 
-void FreeChunk(Chunk *chunk) {
-    free(chunk->samples);
+void FreeChunk(Chunk_t *chunk) {
+    free(((Chunk *)chunk)->samples);
     free(chunk);
 }
 
 static int IsChunkFull(Chunk *chunk) {
     return chunk->num_samples == chunk->max_samples;
 }
 
-int ChunkNumOfSample(Chunk *chunk) {
-    return chunk->num_samples;
+u_int64_t ChunkNumOfSample(Chunk_t *chunk) {
+    return ((Chunk *)chunk)->num_samples;
 }
 
-Sample *ChunkGetSampleArray(Chunk *chunk) {
+static Sample *ChunkGetSampleArray(Chunk *chunk) {
     return (Sample *)chunk->samples;
 }
 
-Sample *ChunkGetSample(Chunk *chunk, int index) {
+static Sample *ChunkGetSample(Chunk *chunk, int index) {
     return &ChunkGetSampleArray(chunk)[index];
 }
 
-timestamp_t ChunkGetLastTimestamp(Chunk *chunk) {
-    if (chunk->num_samples == 0) {
+timestamp_t ChunkGetLastTimestamp(Chunk_t *chunk) {
+    if (((Chunk *)chunk)->num_samples == 0) {
         return -1;
```

Comment:
This can be done in future cleanups since as you noted, CR_ERR != -1

Reply:
Looking further, it should be removed altogether since when this function is called for info it returns `-1` where `0` is a more appropriate return value.
It does not change functionality and passes tests when removed.
But this should be in a separate PR.